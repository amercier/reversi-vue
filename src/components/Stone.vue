<template>
  <div :class="['stone', `stone--${color}`]"></div>
</template>

<script>
import { COLORS } from '../config';

/**
 * A stone on the board.
 */
export default {
  name: 'Stone',
  props: {
    /**
     * Color of the stone.
     *
     * @type {String}
     */
    color: {
      type: String,
      validator: value => COLORS.indexOf(value) !== -1,
      required: true,
    },
  },
};
</script>

<style lang="scss">
$inner-circle-padding: 10%;
$colors: (
  white: (
    #fff,
    #999,
  ),
  black: (
    #292929,
    #090909,
  ),
);

.stone {
  position: relative;
  box-shadow: 0 0 0.5em 0 #000;
  border-radius: 50%;

  &::before {
    content: '';
    position: absolute;
    border-radius: 50%;
    top: $inner-circle-padding;
    bottom: $inner-circle-padding;
    right: $inner-circle-padding;
    left: $inner-circle-padding;
  }

  @each $color, $gradient in $colors {
    &--#{$color} {
      background: linear-gradient(
        to bottom,
        nth($gradient, 1),
        nth($gradient, 2)
      );

      &::before {
        background: linear-gradient(
          to bottom,
          nth($gradient, 2),
          nth($gradient, 1)
        );
      }
    }
  }
}
</style>
