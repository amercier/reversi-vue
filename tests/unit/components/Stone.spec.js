import { shallowMount } from '@vue/test-utils';
import Stone from '@/components/Stone.vue';

describe('Stone.vue', () => {
  it('renders a white stone', () => {
    const stone = shallowMount(Stone, {
      propsData: { color: 'white' },
    });
    expect(stone.html()).toMatchSnapshot();
  });

  it('renders a black stone', () => {
    const stone = shallowMount(Stone, {
      propsData: { color: 'black' },
    });
    expect(stone.html()).toMatchSnapshot();
  });
});
