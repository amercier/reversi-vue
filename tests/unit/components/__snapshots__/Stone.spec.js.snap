// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stone.vue renders a black stone 1`] = `<div class="stone stone--black"></div>`;

exports[`Stone.vue renders a white stone 1`] = `<div class="stone stone--white"></div>`;
